src/Dofs/ConstraintHandler.jl

```diff
@@ -1,64 +1,53 @@
 # abstract type Constraint end
 """
-    DirichletBoundaryConditions
-
-A Dirichlet boundary condition is a boundary where the solution is fixed to take a certain value.
-The struct `DirichletBoundaryConditions` represents a collection of such boundary conditions.
-
-It is created from a `DofHandler`
-
-```jldoctest dbc
-julia> dbc = DirichletBoundaryConditions(dh)
-```
-
-Dirichlet boundary conditions are added to certain components of a field for a specific nodes of the grid.
-A function is also given that should be of the form `(x,t) -> v` where `x` is the coordinate of the node, `t` is a
-time parameter and `v` should be of the same length as the number of components the bc is applied to:
-
-```jldoctest
-julia> addnodeset!(grid, "clamped", x -> norm(x[1]) ≈ 0.0);
-
-julia> nodes = grid.nodesets["clamped"]
-
-julia> push!(dbc, :temperature, nodes, (x,t) -> t * [x[2], 0.0, 0.0], [1, 2, 3])
+    Dirichlet(u, ∂Ω, f)
+    Dirichlet(u, ∂Ω, f, component)
+
+Create a Dirichlet boundary condition on `u` on the `∂Ω` part of
+the boundary. `f` is a function that takes two arguments, `x` and `t`
```

Comment:
Actually, I guess `f` could be just a function of `x` and optionally you could close over a `t` if you wanted. This means `update!` wouldn't take a time either, you would just update `t`. Can do this later though, if we want.